### DISCUSSION TITLE
v2.3.8988

### DISCUSSION THREAD

**Discussion by gbakeman** [nutdotnet/WinNUT-Client#179] [Announcements]
## What's Changed
* Update Overhaul Part 1 - Octokit Library in https://github.com/nutdotnet/WinNUT-Client/pull/176
  * WinNUT updater code overhauled in anticipation of further changes in WinNUT. The Updater GUI looks and functions the same for the most part. In need of translations (although you may want to wait for the next part of this update before making translations.)
  * Added a third party library (Octokit) in favor of custom update query code.
  * Removed the _Auto update_ option since WinNUT doesn't have the functionality to check for an update periodically during operation, it can only check at startup under a different option. See #175 for more info.
  * Please keep in mind that the impacts of these changes won't be completely in force until the _following_ update. I've tested this a few times on my own computer, but the ultimate test won't be until later.

### Build automation
* Modify MSBuild command, version upgrades in https://github.com/nutdotnet/WinNUT-Client/pull/178

**Full Changelog**: https://github.com/nutdotnet/WinNUT-Client/compare/v2.3.8981...v2.3.8988

<hr /><em>This discussion was created from the release <a href='https://github.com/nutdotnet/WinNUT-Client/releases/tag/v2.3.8988'>v2.3.8988</a>.</em>

**Comment by buzzwebly** [comment]
FYI, Windows Defender is currently flagging and quarantining this release as containing a virus (wacatc.b!l)

**Comment by buzzwebly** [comment]
Since you mentioned using new libraries, I'm reluctant to override the quarantine.

**Reply by gbakeman** [reply]
I actually noticed that too, for an older installer I had in my recycle bin. I've just scanned both the latest MSI and ZIP archive, and Windows Defender seemed to think they're OK now. [Here's](https://www.virustotal.com/gui/file/d5bb39fc4419a3db1ae12d35d88c7f3fdd93217cf585f824c139dd69733a4ca6) a VirusTotal scan on the MSI, and [another](https://www.virustotal.com/gui/file/393306c3092dd139424b24451dcaca38f9cac0492421601c734d3e6b92dcdc6b?nocache=1) of the ZIP archive. Save for one false positive in the archive, all appears to be clear. Defender may have thrown a fit because I recently started including the Octokit library (for interacting with GitHub releases.) Hope this helps clear some things up!

**Comment by kamel13** [comment]
When clicking on Settings Menu, the Settings Window does not appear. If you click several times, the window briefly flashes on the screen before disappearing again. Due to this, it is not possible to perform any configuration settings.

**Reply by gbakeman** [reply]
Hi Kamel,

This is definitely a new issue to me! Could I please ask that you make a [new Q&A discussion topic](https://github.com/nutdotnet/WinNUT-Client/discussions/new?category=q-a), with logs attached, and I'll attempt to help you further?

**Reply by gbakeman** [reply]
https://github.com/nutdotnet/WinNUT-Client/discussions/179#discussioncomment-12948216

**Comment by AGenius** [comment]
This problem is a common issue with many apps that remember the forms screen position,  when the settings form is open, float the mouse pointer over the task bar area where the form is showing, the windows for the app will show,  just right click the settings form and choose "Move", then press one of the keyboard arrows to make the form move into view and stick to the mouse pointer.
Once in view , left click to let go.

See if that works.

**Comment by messal14** [comment]
Ich finde es sehr gut, dass jemand das ursprüngliche Projekt wieder aufgegriffen hat. Leider muss ich sagen, dass die hier angebotene Version nicht zu 100% mit Geräten von APC zurechtkommt. Einige Graphen zeigen keine, oder falsche Werte an. So habe ich z.B. an zwei verschiedenen USVs die bei der Batteriespannung keine Werte anzeigen. 
Ich habe es erst als Update über die Ursprungsversion installiert. 
Um hier einen Fehler auszuschließen der durch das Update verursacht werden konnte habe ich es auch mit einer Neuinstallation versucht. Das Ergebnis ist dasselbe. 
Getestet wurden eine Smart-UPS 750 (älteres Model) und eine SMT750RMI2UC, jeweils an verschiedenen Synology-NAS. Verbindungsfehler zu den Servern kann ich ausschließen die Ursprungsversion Fehlerfrei läuft und jetzt auch wieder laufen.

**Reply by gbakeman** [reply]
> I think it's great that someone has revived the original project. Unfortunately, I have to say that the version offered here doesn't fully work with APC devices. Some graphs display no values ​​or incorrect values. For example, I have two different UPSs that don't display any values ​​for battery voltage.
I first installed it as an update over the original version.
To rule out an error caused by the update, I also tried a clean install. The result is the same.
I tested a Smart-UPS 750 (older model) and an SMT750RMI2UC, each connected to different Synology NAS devices. I can rule out connection errors to the servers; the original version is running flawlessly and is now working again.

Hallo @messal14 ,

Ich entschuldige mich im Voraus, wenn ich mich nicht klar ausdrücke, da ich durch einen Übersetzer schreibe.

Es hört sich so an, als ob das Problem, das Sie haben, mit [diesem Problem](https://github.com/nutdotnet/WinNUT-Client/issues/116) zusammenhängt, und Sie sollten vielleicht auch [diese Suchanfrage](https://github.com/search?q=repo%3Anutdotnet%2FWinNUT-Client+apc&type=discussions) überprüfen, da mehrere andere Leute über Probleme mit den APC-USVs berichtet haben. Die kurze Antwort ist, dass WinNUT Standardwerte (Nominalwerte) für wichtige Variablen ersetzt, wenn ein Fehler beim Abrufen dieser Variablen vom NUT-Server auftritt. Es ist sehr wahrscheinlich, dass Ihr NUT-Server diese fehlenden Werte von Anfang an nicht korrekt gemeldet hat, und erst als das Verhalten in WinNUT geändert wurde, ist es offensichtlich geworden, dass etwas nicht stimmt.

Bitte überprüfen Sie die Suchanfrage und das Problem, das ich oben verlinkt habe. Wenn diese beiden Dinge Ihre Frage nicht beantworten, zögern Sie bitte nicht, [eine neue Q&A-Diskussion zu eröffnen](https://github.com/nutdotnet/WinNUT-Client/discussions/new?category=q-a) mit wichtigen Details wie Debug-Protokollen und der Ausgabe von NUT-Servervariablen, und ich werde mein Bestes tun, um zu helfen.

Ich danke Ihnen für Ihre Unterstützung und hoffe, dass wir einen Weg finden, die richtigen Werte für Sie zu ermitteln.

====

Hello @messal14 ,

I apologize in advance if I don't come through clearly as I'm writing through a translator.

It sounds like the issue you're experiencing is related to [this issue](https://github.com/nutdotnet/WinNUT-Client/issues/116), and you may want to check [this search query](https://github.com/search?q=repo%3Anutdotnet%2FWinNUT-Client+apc&type=discussions) as several other people have reported problems with the APC UPSs. The short answer is that WinNUT used to substitute default (nominal) values in for important variables if an error is encountered while retrieving that variable from the NUT server. It's very likely that your NUT server was not reporting those missing values correctly from the beginning, and only when the behavior was changed in WinNUT did it become apparent that something was wrong.

Please check that search query and issue I linked above. If those two things don't answer your question, please feel free to [open a new Q&A discussion](https://github.com/nutdotnet/WinNUT-Client/discussions/new?category=q-a) with important details such as debug logs and NUT server variable output, and I'll do my best to help.

Thank you for your support, and I hope we can find a way to get the correct values for you.
